### PR TITLE
Inbound/Outbound Router

### PR DESCRIPTION
**Pull request by nczeranka**
Kafka events used for tx-Context enrichment

### CODE REVIEW

File: src/server/events/TransactionContext.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ *          serviceProfile: {
+ *              id: 4567
+ *              clientId: ...
+ *              ...
+ *          }
+ *      }
+ *      ...
+ * }
+ */
+module.exports.enrichWithServiceProfile = async function(tx, context, topic, logger, eventClient, db)
```
Comment (toreskogp) [inline src/server/events/TransactionContext.js]: I fell this is a bit many arguments for a function. What about using a class with a constructor for initializing the logger, event client and db?

File: src/server/events/TransactionContext.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+{
+    logger.info("enrichTxWithServiceProfile - payload, context, topic: ", tx, context, topic);
+
+    const type = topic.match(/inbound/) ? "inbound" : "outbound";
+    const typeName = type[0].toUpperCase() + type.slice(1);
+
+    try
+    {
+        const apiServiceProfile = new ServiceProfile(db);
```
Comment (toreskogp) [inline src/server/events/TransactionContext.js]: This line could probably have been in the constructor together with the initialization of the logger, event client, and the db.

File: src/server/events/TransactionContext.js (hunk 1 above)
Comment (nczeranka) [reply src/server/events/TransactionContext.js]: Will revise the usage with the upcoming adjustments (ReceptionChannelConfig as part of the ServiceProfile, tnt-logging v2, ...)

File: src/server/events/TransactionContext.js (hunk 2 above)
Comment (nczeranka) [reply src/server/events/TransactionContext.js]: Will revise the usage with the upcoming adjustments (ReceptionChannelConfig as part of the ServiceProfile, tnt-logging v2, ...)

**Pull request merged**